skyrim-platform/src/platform_se/skyrim_platform/GameEventSinks.cpp

```diff
@@ -665,3 +732,1297 @@ RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+
+      if (weaponAlchemyFlag) {
+        EventsApi::SendEvent("itemCrafted", { JsValue::Undefined(), obj });
+      }
+      if (enchantingFlag) {
+        EventsApi::SendEvent("itemEnchanted", { JsValue::Undefined(), obj });
+      }
+    });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const RE::Inventory::Event* event,
+  RE::BSTEventSource<RE::Inventory::Event>* eventSource)
+{
+  if (event == nullptr) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto refr = event->objRefr;
+  auto refrId = refr ? refr->formID : 0;
+  auto entryData = event->entryData;
+  auto newCount = event->newCount;
+  auto prevCount = event->prevCount;
+  auto itemName = entryData->GetDisplayName();
+  auto itemOwner = entryData->GetOwner();
+  auto itemOwnerId = itemOwner ? itemOwner->formID : 0;
+  auto itemValue = entryData->GetValue();
+  auto itemWeight = entryData->GetWeight();
+  auto itemSoulLevel = static_cast<int>(entryData->GetSoulLevel());
+
+  if (itemName == nullptr || itemValue < 0 || itemWeight < 0 ||
+      itemSoulLevel < 0 || itemSoulLevel > 5 || newCount < 0 ||
+      prevCount < 0) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask(
+    [refr, refrId, newCount, prevCount, itemName, itemOwner, itemOwnerId,
+     itemValue, itemWeight, itemSoulLevel] {
+      auto obj = JsValue::Object();
+
+      auto refrLocal = RE::TESForm::LookupByID(refrId);
+      auto itemOwnerLocal = RE::TESForm::LookupByID(itemOwnerId);
+
+      if (refrLocal == nullptr || refrLocal != refr) {
+        return;
+      }
+
+      obj.SetProperty("reference", CreateObject("ObjectReference", refrLocal));
+      obj.SetProperty("itemName", JsValue::String(itemName));
+      if (itemOwnerLocal != nullptr || itemOwner == itemOwnerLocal) {
+        obj.SetProperty("itemOwner",
+                        CreateObject("ObjectReference", itemOwnerLocal));
+      }
+      obj.SetProperty("itemValue", JsValue::Double(itemValue));
+      obj.SetProperty("itemWeight", JsValue::Double(itemWeight));
+      obj.SetProperty("itemSoulLevel", JsValue::Double(itemSoulLevel));
+      obj.SetProperty("prevCount", JsValue::Double(prevCount));
+      obj.SetProperty("newCount", JsValue::Double(newCount));
+
+      EventsApi::SendEvent("inventory", { JsValue::Undefined(), obj });
+    });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const RE::ChestsLooted::Event* event,
+  RE::BSTEventSource<RE::ChestsLooted::Event>* eventSource)
+{
+  if (event == nullptr) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask([] {
+    auto obj = JsValue::Object();
+
+    EventsApi::SendEvent("chestLooted", { JsValue::Undefined(), obj });
+  });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
```

Comment:
this also cannot be merged until we will be able to access pickpocketed actor and item. remove from this PR for now.